### PR TITLE
Better simulate problematic plugins permissions in unit tests.

### PR DESCRIPTION
**Pull request by rmuir**
We don't have a plugin .zip for unit tests, so we can't do it
correctly. But we can approximate it better, so that if code
is simply missing an AccessController block at least tests will fail.

Cloud-gce unit tests now fail, as they should: they are missing some blocks
around problematic GCE code but we don't yet have the integ tests to catch them.
David only found this out via manual testing on https://github.com/elastic/elasticsearch/pull/13612

I will merge in his changes (to separate out from that unrelated PR) and add anything else missing until unit tests are passing.

Currently the plugins in question are cloud ones, and so we need to 
get the best we can out of the tests we have.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -279,7 +280,7 @@
                                 <include>rest-api-spec/**/*</include>
                                 <include>org/elasticsearch/test/**/*</include>
                                 <include>org/elasticsearch/bootstrap/BootstrapForTesting.class</include>
-                                <include>org/elasticsearch/bootstrap/XTestSecurityManager*.class</include>
+                                <include>org/elasticsearch/bootstrap/MockPluginPolicy.class</include>
```
Comment (rjernst) [inline core/pom.xml]: doubled up  MockPluginPolicy?

File: core/src/test/java/org/elasticsearch/bootstrap/MockPluginPolicy.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        // every element in classpath except test-classes/
+        extraSources = new HashSet<CodeSource>();
+        for (URL location : JarHell.parseClassPath()) {
+            Path path = PathUtils.get(location.toURI());
+            String baseName = path.getFileName().toString();
+            if (baseName.contains("test-classes") == false) {
+              extraSources.add(new CodeSource(location, (Certificate[])null));
+            }
+        }
+        // exclude some obvious places
```
Comment (rjernst) [inline core/src/test/java/org/elasticsearch/bootstrap/MockPluginPolicy.java]: maybe add a single comment here explaining that every stack frame element must have permissions, thus all the test-framework stuff needs it as well (obvious to someone who knows java security well, non-obvious to someone browsing the code)

**Comment by rjernst** [comment]
LGTM

File: core/pom.xml (hunk 1 above)
Comment (rmuir) [inline core/pom.xml]: its not doubled up: currently these things are listed twice, once for the stuff that creates the sources jar, the other for the binary jar. We should really try to clean that up!

File: core/src/test/java/org/elasticsearch/bootstrap/MockPluginPolicy.java (hunk 2 above)
Comment (rmuir) [inline core/src/test/java/org/elasticsearch/bootstrap/MockPluginPolicy.java]: Does this help? https://github.com/rmuir/elasticsearch/commit/0fdc16927efa89a98285fd3b9ba51d4f1a7cb1f2

Comment (rjernst) [inline core/src/test/java/org/elasticsearch/bootstrap/MockPluginPolicy.java]: Looks good!

**Comment by rmuir** [comment]
I just backported this to 2.x

**Pull request merged**